src/wallet/wallet.cpp

```diff
@@ -1415,11 +1416,25 @@ void CWallet::SetHDChain(const CHDChain& chain, bool memonly)
     hdChain = chain;
 }
 
-bool CWallet::IsHDEnabled() const
+bool CWallet::HasHDSeed() const
 {
     return !hdChain.seed_id.IsNull();
 }
 
+bool CWallet::IsHDEnabled() const
+{
+    LOCK(cs_wallet);
+    // -usehd was removed in 0.16, so all wallets with FEATURE_PRE_SPLIT_KEYPOOL are HD
+    // For older wallets the presence of seed_id indicates if HD is enabled.
+    // Support for empty HD capable wallets was added in 0.18
+    return CanSupportFeature(FEATURE_PRE_SPLIT_KEYPOOL) || HasHDSeed();
+}
+
+bool CWallet::IsBlank() const
+{
+    return IsHDEnabled() && !HasHDSeed();
```

Comment:
Nit: document that for older wallets this relies on `IsHDEnabled()` checking `HasHDSeed()`, so there's no way this returns a false positive.